doc/source/user_guide/plotting_with_matplotlib.md

```diff
@@ -0,0 +1,103 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+(plotting_with_matplotlib)=
+
+# Plotting with Matplotlib
+
+{doc}`Matplotlib <mpl:index>` is the default backend for ArviZ. Although most of the functions in the {ref}`plot <plot_api>` module work seamlessly with any backend, some advanced plots may require the use of backend specific features. In this guide, advanced plotting with Matplotlib will be covered.
+
+This page can be downloaded as a {jupyter-download:script}`Python script <plotting_with_matplotlib>`
+or as a {jupyter-download:nb}`Jupyter notebook <plotting_with_matplotlib>`.
+
+```{code-cell} ipython3
+import arviz as az
+import matplotlib.pyplot as plt
+import numpy as np
+
+az.style.use("arviz-darkgrid")
+```
+
+## Customizing plots
+
+### Using `backend_kwargs`
+
+The `backend_kwargs` argument can be very useful for some specific configuration. That is parameters available in {func}`matplotlib.pyplot.subplots`, which includes {meth}`~matplotlib.figure.Figure.add_subplot` and {class}`~mpl:matplotlib.gridspec.GridSpec` through `subplot_kw` and `gridspec_kw` respectively.
+
+As an example, the following code changes the `facecolor` from {func}`~matplotlib.pyplot.figure` and the `width_ratios` in the grid layout.
+
+```{code-cell} ipython3
+# load data
+data = az.load_arviz_data('radon')
+```
+
+```{code-cell} ipython3
+az.plot_posterior(
+    data,
+    var_names=["g"],
+    backend_kwargs={
+        "facecolor": "#d3d0e3",
+        "gridspec_kw": {
+            "width_ratios": [6,4]}});
+```
+:::{note}
+This parameter is however not as flexible as creating custom axes.
+:::
+:::{caution}
+The parameters `ncol` and `nrows` from {func}`matplotlib.pyplot.subplots` should not be set using the `backend_kwargs` parameter.
+:::
+
+### The parameter `show`
+
+:::{note} The parameter `show` calls the show function of the backend. Matplotlib shows the plot in a Jupyter notebook even if this is set to `False`. For more details see {func}`matplotlib.pyplot.show`.
```

Comment:
```suggestion
The parameter `show` calls {func}`matplotlib.pyplot.show` after all ArviZ plotting finishes. defaults rcparam... (like with bokeh but defaults to false)

:::{note} There are some Matplotlib backends like the inline (default) Jupyter backend that show the plot even if `plt.show` is not called. In such cases, plots will be shown automatically even if `show=False` as this only skips calling `plt.show` it doesn't do anything to prevent the plots from showing.
```

Reply:
I did some changes here. I am not sure if I understood the part about the `rcparam`, I couldn't find any similar  rcparam in matplotlib.

Reply:
Added